taichi/codegen/llvm/codegen_llvm.cpp

```diff
@@ -2639,6 +2639,12 @@ LLVMCompiledTask TaskCodeGenLLVM::run_compilation() {
       TI_ASSERT(func);
       tlctx->mark_function_as_cuda_kernel(func, task.block_dim);
     }
+  } else if (config.arch == Arch::amdgpu) {
+    for (const auto &task : offloaded_tasks) {
+      llvm::Function *func = module->getFunction(task.name);
+      TI_ASSERT(func);
+      func->setCallingConv(llvm::CallingConv::AMDGPU_KERNEL);
+    }
```

Comment:
Is it possible to organize this into a ModulePass in `llvm_context_pass.h`?

Reply:
Yeah. But I think this function that marks the taichi-kernel as a device-side function is simple.
What do you think about packaging it into a `mark_function_as_amdgpu_kernel` like cuda?

Reply:
Yup, I'd vote for making a `mark_function_as_amdgpu_kernel` function